RedfishPkg/RedfishCredentialDxe/RedfishCredentialDxe.c

```diff
@@ -123,6 +130,666 @@ RedfishCredentialStopService (
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: can not remove bootstrap account at BMC: %r", __func__, Status));
+    DumpRedfishResponse (__func__, DEBUG_ERROR, &RedfishResponse);
+  } else {
+    DEBUG (
+      (REDFISH_CREDENTIAL_DEBUG, "%a: bootstrap account: %a is removed from: %s\nURI - %s",
+       __func__, mCredentialPrivate->AccountName, REDFISH_MANAGER_ACCOUNT_COLLECTION_URI, TargetUri)
+      );
+  }
+
+  RedfishHttpFreeResponse (&RedfishResponse);
+
+  return Status;
+}
+
+/**
+  Get the information about specific Account.
+  Checks the User Name and if name matches delete that account
+
+
+  @param[in]  RedfishService       Pointer to Redfish Service to be used
+                                   for sending DELETE request to BMC.
+  @param[in]  AccountUri           URI of bootstrap account to verify.
+
+**/
+BOOLEAN
+ProcessRedfishBootstarpAccount (
+  IN REDFISH_SERVICE  RedfishService,
+  IN EFI_STRING       AccountUri
+  )
+{
+  EDKII_JSON_VALUE  JsonUserName;
+  EDKII_JSON_VALUE  JsonValue;
+  EFI_STATUS        Status;
+  REDFISH_RESPONSE  RedfishResponse;
+  REDFISH_REQUEST   RedfishRequest;
+  BOOLEAN           Ret;
+
+  if (mCredentialPrivate == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a: failed with error - %r\n", __func__, EFI_NOT_STARTED));
+    return FALSE;
+  }
+
+  if ((RedfishService == NULL) || IS_EMPTY_STRING (AccountUri) ||
+      (mCredentialPrivate->AuthMethod != AuthMethodHttpBasic))
+  {
+    return FALSE;
+  }
+
+  ZeroMem (&RedfishResponse, sizeof (REDFISH_RESPONSE));
+  ZeroMem (&RedfishRequest, sizeof (REDFISH_REQUEST));
+  Status = RedfishHttpGetResource (RedfishService, AccountUri, &RedfishRequest, &RedfishResponse, FALSE);
+  if (EFI_ERROR (Status) || (RedfishResponse.Payload == NULL)) {
+    DEBUG ((DEBUG_ERROR, "%a: can not get account from BMC: %r", __func__, Status));
+    DumpRedfishResponse (__func__, DEBUG_ERROR, &RedfishResponse);
+    return FALSE;
+  }
+
+  Ret       = FALSE;
+  JsonValue = RedfishJsonInPayload (RedfishResponse.Payload);
+  if (JsonValueIsObject (JsonValue)) {
+    JsonUserName = JsonObjectGetValue (JsonValueGetObject (JsonValue), "UserName");
+    if (JsonValueIsString (JsonUserName) && (JsonValueGetAsciiString (JsonUserName) != NULL)) {
+      if (AsciiStrCmp (mCredentialPrivate->AccountName, JsonValueGetAsciiString (JsonUserName)) == 0) {
+        DeleteRedfishBootstrapAccount (RedfishService, AccountUri);
+        Ret = TRUE;
+      }
+    }
+  }
+
+  RedfishHttpFreeResponse (&RedfishResponse);
+  RedfishHttpFreeRequest (&RedfishRequest);
+
+  return Ret;
+}
+
+/**
+  This function returns the string of Redfish service version.
+
+  @param[in]   RedfishService       Redfish service instance.
+  @param[out]  ServiceVersionStr    Redfish service string.
+
+  @return     EFI_STATUS
+
+**/
+EFI_STATUS
+RedfishGetServiceVersion (
+  IN  REDFISH_SERVICE  RedfishService,
+  OUT CHAR8            **ServiceVersionStr
+  )
+{
+  *ServiceVersionStr = "v1";

```

Comment:
Should we provide a PCD for platform to define it?  Below is the example in RedfishClientPkg

https://github.com/tianocore/edk2-redfish-client/blob/00062f606c170d4e105b05ecc6ac5f8a3f16b52f/RedfishClientPkg/RedfishClientPkg.dec#L73

Reply:
Agree. I will add that PCD into RedfishPkg.dec